pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,368 @@
+    E: number
+        Beam Energy
+    lens_set : `list`
+        [(numer1, lensthick1), (number2, lensthick2)...]
+    material : `str`
+        Beryllium. The use of beryllium extends the range of operation
+        of compound refractive lenses, improving transmission,
+        aperture size, and gain
+    density : TODO: find out what density is
+
+    Returns
+    -------
+    focal_length : `float`
+    """
+    num = []
+    rad = []
+    ftot_inverse = 0
+    # if type(lens_set) is int:
+    #     lens_set = getLensSet(lens_set)
+    for i in range(len(lens_set) / 2):
+        # TODO: why new code is (double //) range(len(lens_set)//2):
+        num = lens_set[2 * i]
+        rad = lens_set[2 * i + 1]
+        if rad is not None:
+            rad = float(rad)
+            num = float(num)
+            ftot_inverse += num / calcFocalLengthForSingleLens(
+                            E, rad, material, density)
+    return 1.0 / ftot_inverse
+
+
+ def calc_beam_fwhm(self, E, lens_set, distance=None, source_distance=None, material="Be",
+                       density=None, fwhm_unfocused=None, printsummary=True):
+    """
+    Calculates beam parameters for certain lenses configuration
+    and energy at a given distance.
+    Optionally some other parameters can be set
+    Beamsize for certain lenses configuration and energy at a given distance
+
+    Parameters
+    -----------
+    E: number
+        Beam Energy
+    lens_set : `list`
+        [(numer1, lensthick1), (number2, lensthick2)...]
+    distance: float
+        Distance from the lenses to the sample is 3.852 m at XPP.
+    source_distance: float
+        Distance from source to lenses. This is about 160 m at XPP.
+    material: str
+        Beryllium. The use of beryllium extends the range of operation
+        of compound refractive lenses, improving transmission,
+        aperture size, and gain
+    density: TODO: find out what density is
+    fwhm_unfocused: float
+        This is about 400 microns at XPP.
+    printsummary: boolean
+        Prints summary of parameters/calculations if True
+
+    Returns
+    -------
+    float: Size FWHM
+    """
+    # Focal length for certain lenses configuration and energy
+    focal_length = calc_focal_length(E, lens_set, material, density)
+    
+    # use lens makers equation to find distance to image of source
+    if source_distance is not None:
+        focal_length = 1 / (1 / focal_length - 1 / source_distance)
+    
+    lam = 1.2398 / E * 1e-9
+    # the w parameter used in the usual formula is 2*sigma
+    w_unfocused = fwhm_unfocused * 2 / 2.35
+    # assuming gaussian beam divergence = w_unfocused/f we can obtain
+    waist = lam / np.pi * focal_length / w_unfocused
+    rayleigh_range = np.pi * waist ** 2 / lam
+    size = waist * np.sqrt(1.0 + (distance - focal_length) ** 2.0 / rayleigh_range ** 2)
+    size_fwhm = size * 2.35 / 2.0
+
+    if printsummary:
+        print("FWHM at lens   : %.3e" % (fwhm_unfocused))
```

Comment:
For stuff like this that'll likely be repeated in many spots, I think we should provide recommended ways of doing things. Perhaps that could be something like using [tabulate](https://pypi.org/project/tabulate/) to create tables?

Reply:
I'll give that a try, I think that's a great idea, thank you